sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollerTests.java

```diff
@@ -221,113 +145,50 @@ public void run() {
-            createPollOperation(inProgressPollResponse,
-                successPollResponse, totalTimeoutInMillis / 2);
+        when(pollOperation.apply(any())).thenReturn(Mono.just(inProgressPollResponse), Mono.just(successPollResponse));
 
-        Poller<CreateCertificateResponse> createCertPoller = new Poller<>(pollInterval, pollOperation);
+        Poller<Response> createCertPoller = new Poller<>(pollInterval, pollOperation);
 
         while (createCertPoller.getStatus() != OperationStatus.SUCCESSFULLY_COMPLETED) {
             Thread.sleep(pollInterval.toMillis());
         }
 
-        Assert.assertTrue(createCertPoller.getStatus() == OperationStatus.SUCCESSFULLY_COMPLETED);
-        Assert.assertTrue(createCertPoller.isAutoPollingEnabled());
-    }
-
-    /* Test where SDK Client is subscribed all responses.
-     * The last response in this case will be PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
-     * This scenario is setup where source will generate successful response returned after few in-progress response.
-     **/
-    @Test
-    public void subscribeToAllPollEventSuccessfullyCompleteInNSecondsTest() throws Exception {
-
-        PollResponse<CreateCertificateResponse> successPollResponse = new PollResponse<>(PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED, new CreateCertificateResponse("Created : Cert A"));
-        PollResponse<CreateCertificateResponse> inProgressPollResponse = new PollResponse<>(PollResponse.OperationStatus.IN_PROGRESS, new CreateCertificateResponse("Starting : Cert A"));
-
-        long totalTimeoutInMillis = 1000 * 1;
-        Duration pollInterval = Duration.ofMillis(totalTimeoutInMillis / 10);
-
-        Function<PollResponse<CreateCertificateResponse>, Mono<PollResponse<CreateCertificateResponse>>> pollOperation =
-            createPollOperation(inProgressPollResponse,
-                successPollResponse, pollInterval.toMillis() * 2);
-
-        Poller<CreateCertificateResponse> createCertPoller = new Poller<>(pollInterval, pollOperation);
-        Thread.sleep(totalTimeoutInMillis);
-        debug("Calling poller.block ");
-        Assert.assertTrue(createCertPoller.block().getStatus() == OperationStatus.SUCCESSFULLY_COMPLETED);
-        Assert.assertTrue(createCertPoller.getStatus() == OperationStatus.SUCCESSFULLY_COMPLETED);
-        Assert.assertTrue(createCertPoller.isAutoPollingEnabled());
-    }
-
-    /* Test where SDK Client is subscribed to only final/last response.
-     * The last response in this case will be PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
-     * This scenario is setup where source will generate successful response returned after few in progress response.
-     * But the subscriber is only interested in last response, The test will ensure subscriber
-     * only gets last PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED . */
-
-    @Test
-    public void subscribeToOnlyFinalEventSuccessfullyCompleteInNSecondsTest() throws Exception {
```

Comment:
Why are we removing this test ? 
This is one of the usecase which was part mentioned in requirements. Are we testing this somewhere else ?
`subscriber is only interested in last response`

Reply:
Yes we are. In all the other tests, we ensure we get the last response.

Reply:
I'll add this one back.